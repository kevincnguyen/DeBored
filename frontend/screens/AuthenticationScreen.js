import { View, Text, StyleSheet } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { Button } from "react-native-paper";
import { useSafeAreaInsets } from "react-native-safe-area-context";

const AuthenticationScreen = () => {
  const insets = useSafeAreaInsets();
  const navigation = useNavigation();

  const navigateToLogin = () => {
    navigation.navigate("Login");
  };

  const navigateToSignup = () => {
    navigation.navigate("SignUp");
  };

  return (
    <View
      style={[
        styles.container,
        { paddingTop: insets.top + 100, paddingBottom: insets.bottom + 100 },
      ]}
    >
      <View style={styles.hookContainer}>
        <View style={styles.headingContainer}>
          <Text style={styles.title}>Bored?</Text>
          <Text style={styles.slogan}>DeBored</Text>
        </View>
        <Text style={styles.text}>Fight boredom. Make friends.</Text>
      </View>
      <View>
        <Button
          mode="contained"
          onPress={navigateToLogin}
          contentStyle={styles.buttonContent}
          labelStyle={styles.text}
          style={styles.button}
        >
          Login
        </Button>
        <Button
          mode="contained-tonal"
          onPress={navigateToSignup}
          contentStyle={styles.buttonContent}
          labelStyle={styles.text}
          style={styles.button}
        >
          Sign up
        </Button>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "space-between",
    alignItems: "center",
    backgroundColor: "#FFFFFF",
  },
  hookContainer: {
    flex: 1,
    flexDirection: "column",
    alignItems: "center",
  },
  headingContainer: {
    width: "80%",
  },
  title: {
    fontSize: 55,
    fontWeight: "bold",
  },
  slogan: {
    fontSize: 55,
    fontWeight: "bold",
    color: "#7845AC",
  },
  button: {
    marginBottom: 10,
    borderRadius: 10,
  },
  buttonContent: {
    paddingVertical: 10,
    width: 350,
  },
  text: {
    fontSize: 18,
  },
});

export default AuthenticationScreen;
