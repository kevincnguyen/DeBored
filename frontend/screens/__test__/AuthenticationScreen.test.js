import React from "react";
import renderer from "react-test-renderer";
import { Button } from "react-native-paper";
import { useNavigation } from "@react-navigation/native";
import { useSafeAreaInsets } from "react-native-safe-area-context";

import AuthenticationScreen from "../AuthenticationScreen";

// Mock the external modules
jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(),
}));
jest.mock("react-native-safe-area-context", () => ({
  useSafeAreaInsets: jest.fn(),
}));

test("renders AuthenticationScreen component correctly", () => {
  useSafeAreaInsets.mockImplementation(() => ({ top: 0, bottom: 0 }));
  const tree = renderer.create(<AuthenticationScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("displays correct text elements", () => {
  useSafeAreaInsets.mockImplementation(() => ({ top: 0, bottom: 0 }));
  const tree = renderer.create(<AuthenticationScreen />).toJSON();
  expect(JSON.stringify(tree)).toMatch("Bored?");
  expect(JSON.stringify(tree)).toMatch("DeBored");
  expect(JSON.stringify(tree)).toMatch("Fight boredom. Make friends.");
});

test("navigates to Login and SignUp screens on button presses", () => {
  const mockNavigate = jest.fn();
  useNavigation.mockImplementation(() => ({ navigate: mockNavigate }));
  useSafeAreaInsets.mockImplementation(() => ({ top: 0, bottom: 0 }));

  const component = renderer.create(<AuthenticationScreen />);
  const root = component.root;

  // Simulating Login button press
  root.findAllByType(Button)[0].props.onPress();
  expect(mockNavigate).toHaveBeenCalledWith("Login");

  // Simulating SignUp button press
  mockNavigate.mockClear();
  root.findAllByType(Button)[1].props.onPress();
  expect(mockNavigate).toHaveBeenCalledWith("SignUp");
});
