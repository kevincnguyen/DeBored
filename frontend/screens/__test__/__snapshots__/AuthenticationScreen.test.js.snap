// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AuthenticationScreen component correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "space-between",
      },
      {
        "paddingBottom": 100,
        "paddingTop": 100,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        {
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 55,
            "fontWeight": "bold",
          }
        }
      >
        Bored?
      </Text>
      <Text
        style={
          {
            "color": "#7845AC",
            "fontSize": 55,
            "fontWeight": "bold",
          }
        }
      >
        DeBored
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 18,
        }
      }
    >
      Fight boredom. Make friends.
    </Text>
  </View>
  <View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "borderRadius": 10,
          "marginBottom": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderColor": "transparent",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 10,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "paddingVertical": 10,
                  "width": 350,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "fontSize": 18,
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(232, 222, 248, 1)",
          "borderRadius": 10,
          "marginBottom": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(232, 222, 248, 1)",
            "borderColor": "transparent",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 10,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "paddingVertical": 10,
                  "width": 350,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(29, 25, 43, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "fontSize": 18,
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              Sign up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
